lib/OpenQA/WebAPI/Controller/API/V1/Locks.pm

```diff
@@ -18,31 +18,44 @@ use Mojo::Base 'Mojolicious::Controller';
 
 use OpenQA::IPC;
 
-sub mutex_lock {
-    my ($self) = @_;
-    my $name   = $self->stash('name');
-    my $jobid  = $self->stash('job_id');
-    my $ipc    = OpenQA::IPC->ipc;
-    my $res = $ipc->scheduler('mutex_lock', $name, $jobid);
-    return $self->render(text => 'ack', status => 200) if $res;
-    return $self->render(text => 'nack', status => 409);
-}
+sub mutex_action {
+    my ($self)     = @_;
+    my $name       = $self->stash('name');
+    my $jobid      = $self->stash('job_id');
+    my $validation = $self->validation;
 
-sub mutex_unlock {
-    my ($self) = @_;
-    my $name   = $self->stash('name');
-    my $jobid  = $self->stash('job_id');
+    $validation->required('action')->in(qw(lock unlock));
+    $validation->optional('where')->like(qr/^[0-9]+$/);
```

Comment:
In public API there should be only job_id or nothing (which means the default: check self or parent).
Otherwise the check for finished jobs won't work.